### ISSUE TITLE
page.waitForSelector should return an ElementHandle

### ISSUE THREAD

**Issue by ebidel** [puppeteer/puppeteer#1007]
It would be convenient for `page.waitForSelector` to return the element it's waiting for.

Consider:

```js
await page.waitForSelector('.tweet.permalink-tweet');
const overlay = await page.$('.tweet.permalink-tweet');
await overlay.screenshot({path: 'tweet.png'});
```

could be reduced to: 

```js
const overlay = await page.waitForSelector('.tweet.permalink-tweet');
await overlay.screenshot({path: 'tweet.png'});
```

**Comment by entrptaher** [comment]
On this line, https://github.com/GoogleChrome/puppeteer/blob/master/lib/FrameManager.js#L382

I did this,

```js
const elemVisible = this.waitForFunction(predicate, {timeout, polling}, selector, waitForVisible);
return this.$(selector);
```

I'm not sure about side effects.
Looks like I messed up :dagger:

**Comment by sukrosono** [comment]
I also notice that the waitForSelector did not accept `0` timeout (disable timeout). I do goto with disable timeout but the watForSelector was rejected with 30sec timeout.

**Comment by ebidel** [comment]
Can this make it in 1.0?

**Comment by ebidel** [comment]
Ping. 

Users are still [writing code](https://github.com/GoogleChrome/puppeteer/issues/1546#issue-280072932) that could be DRY.

**Comment by sukrosono** [comment]
I am not able to write some code due to my hdd failing. I like to suggest giving label; maybe feature, P1. With that folks who are interested to contribute will easily find important issue.

**Comment by aslushnikov** [comment]
This is fixed in 1.0: https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectorselector-options
